prow/gerrit/adapter/adapter.go

```diff
@@ -506,6 +508,23 @@ func (c *Controller) handleInRepoConfigError(err error, instance string, change
 	return nil
 }
 
+// shouldSkipProcessingChange determines if we chan skip processing the change when there is no new commit or relavent commands in the comment messages
+func (c *Controller) shouldSkipProcessingChange(instance string, change client.ChangeInfo) bool {
+	lastUpdate, exists := c.tracker.Current()[instance][change.Project]
+	if !exists {
+		lastUpdate = time.Now()
+	}
+
+	for _, message := range currentMessages(change, lastUpdate) {
+		if pjutil.TestRe.MatchString(message.Message) || pjutil.RetestRe.MatchString(message.Message) || pjutil.TestAllRe.MatchString(message.Message) {
+			return false
+		}
+	}
+
+	revision := change.Revisions[change.CurrentRevision]
+	return !revision.Created.Before(lastUpdate)
```

Comment:
nit: We should probably do this cheap check before looping over all recent comments and doing regexp matching.

Reply:
The logic would be different if we do that since we only update based on revision, but commands can come later than last revision.

Reply:
I don't follow. Can you explain what you mean by "we only update based on revision"? It should be possible to do this cheap check first. Either a new patchset or a triggering commands can cause jobs to be triggered so we just check if the patchset is new and if it is we return false, otherwise we go on to check if there are triggering commands in the comments.

Also this appears to be backwards. We should skip processing changes if the the current revision was created before the last sync (e.g. it has already been processed), right?
Please add some unit tests for this to help catch that.

Reply:
Done. Will add some unit tests.